### PR TITLE
Feature/ocsp changes

### PR DESCRIPTION
**Pull request by rivol**
- Make idcard auth middleware (esp rgs OCSP checks) more flexible
- Add all valid intermediate certs
- Update tox and Travis config
- Update Django/Python requirements

### CODE REVIEW

File: esteid/middleware.py [hunk 1]
```diff
@@ -69,9 +119,38 @@ def get_ocsp_responder_certificate_path(cls):
...
+
+
+class NginxIdCardMiddleware(BaseIdCardMiddleware):
+    DN_HEADER_NAME = 'HTTP_X_CLIENT'
+
+    # TODO: Set to a valid value once we add certificates support
+    CERTIFICATE_HEADER_NAME = None
+
+
+class IdCardMiddleware(NginxIdCardMiddleware):
```
Comment (Jyrno42) [inline esteid/middleware.py]: Add an __init__ for this which calls `warnings.warn`, the message could be something like: `esteid.middleware.IdCardMiddleware is deprecated. Please use ApacheIdCardMiddleware or NginxIdCardMiddleware instead.

File: esteid/middleware.py [hunk 2]
```diff
@@ -69,9 +119,38 @@ def get_ocsp_responder_certificate_path(cls):
...
 
     @classmethod
-    def verify_ocsp(cls, request):
-        issuer = request.META.get('HTTP_X_ISSUER', None)
-        certificate = request.META.get('HTTP_X_CLIENTCERT', None)
+    def verify_ocsp(cls, certificate, issuer):
+        """ Runs OCSP verification and returns error code - 0 means success
+        """
+
         return OCSPVerifier(certificate, issuer,
```
Comment (Jyrno42) [inline esteid/middleware.py:148]: What do you think about padding the ocsp error numbers as well?

File: esteid/middleware.py [hunk 3]
```diff
@@ -69,9 +119,38 @@ def get_ocsp_responder_certificate_path(cls):
...
+
+    def parse_client_dn(self, dn):
+        return parse_rfc_dn(dn)
+
+
+class NginxIdCardMiddleware(BaseIdCardMiddleware):
+    DN_HEADER_NAME = 'HTTP_X_CLIENT'
+
+    # TODO: Set to a valid value once we add certificates support
+    CERTIFICATE_HEADER_NAME = None
```
Comment (Jyrno42) [inline esteid/middleware.py]: No cert support for nginx? Will this be added in the future? (add an issue if so)

**Review by Jyrno42 — CHANGES_REQUESTED** [review]

File: esteid/middleware.py (hunk 2 above)
Comment (rivol) [reply esteid/middleware.py:148]: Not sure about padding, how about creating an enum of possible error codes and returning values of that instead? Would also make checks more readable.

Expanding on this idea, other error codes ala `ERROR_NO_DN` could be similarily converted into enum.
Your thoughts?

Comment (Jyrno42) [reply esteid/middleware.py:148]: How would ssl errors coming from nginx/apache (which are numeric openssl error codes, usually padded) fit into the enums idea?

Comment (rivol) [reply esteid/middleware.py:148]: It probably wouldn't then. Ok, how would you do the OCSP padding? Add... 4000 to them?

Comment (Jyrno42) [reply esteid/middleware.py:148]: Yup

**Review by Jyrno42 — APPROVED** [review]

**Pull request merged**